Методы на проверку

VadyaTheFirst opened the pull request.

VadyaTheFirst committed:
Методы на проверку

Bl3cster commented on src/Main.java:
```diff
@@ -1,5 +1,67 @@
+    public static void verifyHighYear(int year) {
+        boolean divide4 = ((year % 4) == 0);
+        boolean divide100 = ((year % 100) == 0);
+        boolean divide400 = ((year % 400) == 0);
```
лишние переменные

VadyaTheFirst replied:
Вот очень интересно: почему, когда задача была решена в предыдущем уроке эти переменные лишними не были, а когда свел текущую задачу к предыдущей они, вдруг, появились?

Bl3cster commented on src/Main.java:
```diff
@@ -1,5 +1,67 @@
+        } else if (divide100) {
+            System.out.println(year + " год не является високосным");
+        } else if (!(divide4)) {
+            System.out.println(year + " год не является високосным");
```
можно проще и читабельнее прописать

VadyaTheFirst replied:
Прошу написать подробнее ответ. Можно лучше, красивее и.т.д, с моей точки зрения, является очень слабоформализованными терминами

Bl3cster replied:
можно создать всего одно условие. если да, то високосный, иначе - не високосный

VadyaTheFirst replied:
Небольшая справка: високосным является каждый четвертый год, но не является каждый сотый. Также високосным является каждый четырехсотый год.

Вот можно было проще автору задания написать: каждый 4й является високосным и все..
Лан, тут конструкторы классов и вызов методов других классов - основная заноза

Bl3cster replied:
поэтому и условие выходит (year % 4 == 0 && year % 100 != 0 || year % 400 ==0)

VadyaTheFirst replied:
Блин, не понял с первого раза, что ты имеешь в виду...

VadyaTheFirst replied:
Запушил

Bl3cster commented on src/Main.java:
```diff
@@ -1,5 +1,67 @@
+
+
+    public static void verifyVersion(byte clientOS, int clientDeviceYear) {
+        boolean verify = (clientOS == 0);
```
не надо лишних переменных

Bl3cster commented on src/Main.java:
```diff
@@ -1,5 +1,67 @@
+        boolean shortTime = (deliveryDistance <= 20);
+        boolean middleTime = (deliveryDistance > 20 && deliveryDistance <= 60);
+        boolean longTime = (deliveryDistance > 60 && deliveryDistance <= 100);
+
```
Аналогично

Bl3cster commented on src/Main.java:
```diff
@@ -1,5 +1,67 @@
+        } else if (middleTime) {
+            System.out.println("Потребуется дней: " + term);
+        } else if (longTime) {
+            System.out.println("Потребуется дней: " + term);
```
тогда не понятно: зачем нужны эти условия, если решение одинаково.

VadyaTheFirst replied:
Потому, что условия задачи были в предыдущих уроках создать несколько условий

VadyaTheFirst committed:
Методы на проверку
С учетом оптимизации кода.
НО! Ребята, не пишите тогда, что можно использовать решение из предыдущих уроков.
В предыдущих уроках как раз было требования на создание лишних переменных

VadyaTheFirst committed:
Методы на проверку
С учетом оптимизации кода.
НО! Ребята, не пишите тогда, что можно использовать решение из предыдущих уроков.
В предыдущих уроках как раз было требования на создание лишних переменных

VadyaTheFirst committed:
Методы на проверку
С учетом оптимизации кода.
НО! Ребята, не пишите тогда, что можно использовать решение из предыдущих уроков.
В предыдущих уроках как раз было требования на создание лишних переменных
Если устройтво младше текущего года, то система рекомендует установить лайт-версию

VadyaTheFirst committed:
Методы на проверку
С учетом оптимизации кода.
НО! Ребята, не пишите тогда, что можно использовать решение из предыдущих уроков.
В предыдущих уроках как раз было требования на создание лишних переменных
Если устройтво младше текущего года, то система рекомендует установить лайт-версию
